test/e2e/storage/testsuites/provisioning.go

```diff
@@ -306,6 +316,95 @@ func TestDynamicProvisioning(t StorageClassTest, client clientset.Interface, cla
 	return pv
 }
 
+// PVWriteReadCheck checks that a PV retains data.
+//
+// It starts these pods:
+// - The first pod writes 'hello word' to the /mnt/test (= the volume) on one node.
+// - The second pod runs grep 'hello world' on /mnt/test on the same node.
+// - Optional: a third pod runs grep 'hello world' on /mnt/test on a different node.
+//
+// Nodes are selected like this:
+// - the first node is based on nodeName (if not empty) or nodeSelector
+//   (otherwise), random if both are empty
+// - the second node is only used if nodeSelector is set and matches more than
+//   the first node
+// - if both nodeName and nodeSelector are unset, then the test cannot lock
+//   the pod onto the same node
+//
+// If all succeeds, Kubernetes actually allocated something that is
+// persistent across pods and (optional) across nodes.
+//
+// This is a common test that can be called from a StorageClassTest.PvCheck.
+func PVWriteReadCheck(client clientset.Interface, claim *v1.PersistentVolumeClaim, volume *v1.PersistentVolume, nodeName string, nodeSelector map[string]string) {
```

Comment:
Instead of modifying the behavior of the existing check, can we add a new check that forces scheduling on a different node?

My thought is that this should be an explicit new test case.  And that the test case should not have different behavior depending on the driver, ie for hostpath driver, this test case is actually not going to test what it wants to test since hostpath only works on a single node.

Reply:
So in other words, have one test for single node and another test for multi-node?

What I don't like about the current test is that it's non-deterministic on a multi-node cluster. The second pod may or may not end up on the same node.

Reply:
Yes, I think we should have explicit test cases for the two scenarios.

The current test is non-deterministic, however, it is more realistic to what a user would experience.

Reply:
I'm not convinced that multiple test cases is really desirable. If we really wanted to focus each test on a single aspect, we would need three tests
- "can provision a PV"
- "can write/read on single node"
- "can write/read on different nodes"

If already provisioning fails, there's no point in testing the other two cases. If the (presumably simpler) write/read on a single node fails, that should be investigated first. But I see that this is a bit subjective.

BTW, the write/read test was extended to also test mount flags. Should that have been a separate test, too?

Anyway, I have split it up into two separate cases. Scheduling is based entirely on node and pod affinity now, at least for the first pod.

@msau42 please have another look